user/super/com/google/gwt/emul/java/util/Map.java

```diff
@@ -35,6 +36,220 @@
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> of(
+      K k1, V v1,
+      K k2, V v2,
+      K k3, V v3,
+      K k4, V v4,
+      K k5, V v5,
+      K k6, V v6
+  ) {
+    return ofEntries(
+            entry(k1, v1),
+            entry(k2, v2),
+            entry(k3, v3),
+            entry(k4, v4),
+            entry(k5, v5),
+            entry(k6, v6)
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> of(
+      K k1, V v1,
+      K k2, V v2,
+      K k3, V v3,
+      K k4, V v4,
+      K k5, V v5,
+      K k6, V v6,
+      K k7, V v7
+  ) {
+    return ofEntries(
+            entry(k1, v1),
+            entry(k2, v2),
+            entry(k3, v3),
+            entry(k4, v4),
+            entry(k5, v5),
+            entry(k6, v6),
+            entry(k7, v7)
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> of(
+      K k1, V v1,
+      K k2, V v2,
+      K k3, V v3,
+      K k4, V v4,
+      K k5, V v5,
+      K k6, V v6,
+      K k7, V v7,
+      K k8, V v8
+  ) {
+    return ofEntries(
+            entry(k1, v1),
+            entry(k2, v2),
+            entry(k3, v3),
+            entry(k4, v4),
+            entry(k5, v5),
+            entry(k6, v6),
+            entry(k7, v7),
+            entry(k8, v8)
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> of(
+      K k1, V v1,
+      K k2, V v2,
+      K k3, V v3,
+      K k4, V v4,
+      K k5, V v5,
+      K k6, V v6,
+      K k7, V v7,
+      K k8, V v8,
+      K k9, V v9
+  ) {
+    return ofEntries(
+            entry(k1, v1),
+            entry(k2, v2),
+            entry(k3, v3),
+            entry(k4, v4),
+            entry(k5, v5),
+            entry(k6, v6),
+            entry(k7, v7),
+            entry(k8, v8),
+            entry(k9, v9)
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> of(
+      K k1, V v1,
+      K k2, V v2,
+      K k3, V v3,
+      K k4, V v4,
+      K k5, V v5,
+      K k6, V v6,
+      K k7, V v7,
+      K k8, V v8,
+      K k9, V v9,
+      K k10, V v10
+  ) {
+    return ofEntries(
+            entry(k1, v1),
+            entry(k2, v2),
+            entry(k3, v3),
+            entry(k4, v4),
+            entry(k5, v5),
+            entry(k6, v6),
+            entry(k7, v7),
+            entry(k8, v8),
+            entry(k9, v9),
+            entry(k10, v10)
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Entry<K, V> entry(K key, V value) {
+    // This isn't quite consistent with the javadoc, since this is serializable, while entry()
+    // need not be serializable.
+    return new AbstractMap.NonNullableImmutableEntry<>(
+        key,
+        value
+    );
+  }
+
+  @JsIgnore
+  static <K, V> Map<K, V> ofEntries(Entry<? extends K,? extends V>... entries) {
+    Map<K, V> map = new HashMap<>();
+
+    for (int i = 0; i < entries.length; i++) {
+      Entry<? extends K, ? extends V> entry = checkNotNull(entries[i]);
+      K key = entry.getKey();
+      if (entry instanceof AbstractMap.NonNullableImmutableEntry) {
```

Comment:
This is debatable - it might save a bit of time at runtime, but increases output size a bit - the instanceof is never optimized out (the type of the array is never tightened to be NonNullableImmutableEntry[] nor does the compiler recognize that all instances are that specific entry type), so we still have the if check.

Reply:
I can't get this to optimize out cleanly at this time, so I'm going to remove it and replace it with a TODO (as it was in an earlier revision) - if at such a time we can strengthen the compiler's type tightening to handle this case, we can always put it back to generally decrease code size here while maintaining correctness/safety.

Reply:
Reverted.